### PR TITLE
Remote Dataloader

### PR DESCRIPTION
**Pull request by xutongye**
## Description
In normal use case, a Dataset on data owner's machine will not be allowed to get downloaded by data scientist. The DS will only need to access the pointers to sample/batch. So, in this case, we don't need to serde the whole training data. This PR aims at building a RemoteDataset and RemoteDataloader to support this use case in an efficient way. 

## Affected Dependencies
List any dependencies that are required for this change.

## How has this been tested?
- tests/syft/core/remote_dataloader/test_remote_dataloader.py
- example/experimental/tongye/train_boston_with_RemoteDataLoader

## Checklist
- [ ] I have followed the [Contribution Guidelines](https://github.com/OpenMined/.github/blob/master/CONTRIBUTING.md) and [Code of Conduct](https://github.com/OpenMined/.github/blob/master/CODE_OF_CONDUCT.md)
- [ ] I have commented my code following the [OpenMined Styleguide](https://github.com/OpenMined/.github/blob/master/STYLEGUIDE.md)
- [ ] I have labeled this PR with the relevant [Type labels](https://github.com/OpenMined/.github/labels?q=Type%3A)
- [ ] My changes are covered by tests

### CODE REVIEW

**Review by madhavajay — APPROVED** [review]
This works well, I tested the notebook and also modified the RemoteDataset to support csv files and used that in the CDC notebook and it also worked by passing the result of a full slice [:] to the DataFrame constructor.

I think this is sufficient for now so that with a little pre-processing work we can at least load large tensors or dataframes into the store.
